### ISSUE TITLE
data augmentation

### ISSUE THREAD

**Issue by Borda** [PyTorchLightning/lightning-Covid19#6]
Add reasonable image augmentation
- horizontal/vertical flip
- rotation
- zoom
- etc

**Comment by edgarriba** [comment]
cool, we have all this in kornia

**Comment by shpotes** [comment]
Horizontal flip looks like a suitable augmentation, I'm not completely sure if vertical flip/rotation introduces interesting priors as X-rays are usually similarly oriented

**Comment by edgarriba** [comment]
rotation limited to small degrees I guess yes

**Comment by Borda** [comment]
> Horizontal flip looks like a suitable augmentation, I'm not completely sure if vertical flip/rotation introduces interesting priors as X-rays are usually similarly oriented

unless there is the assumption that the object looks vertical different... but this could be just an extra training parameter, right?

**Comment by bluesky314** [comment]
Where is the data in the first place? no link in readme

**Comment by ducha-aiki** [comment]
data link is https://github.com/PyTorchLightning/lightning-Covid19/issues/2  here

**Comment by anguelos** [comment]
In the  [Chester paper](https://arxiv.org/pdf/1901.11210.pdf) In figure 3. We can see that for **pneumonia specifically** augmentation might even do bad.
If I am reading the plots right, the first column (undistorted test set) seems the most important.
It seems that modest rotation scale and translation is the best augmentation. 15deg, 10%, and 10% respectively.

**Comment by shijianjian** [comment]
Generally, I think it should be alright as long as the label will not be changed by augmentation methods. For instance, ElasticTransform is probably a dangerous move. It would be best if we can invite a chest CT expert for more guidance.

If I understand this right, this project aims to tell Covid-19 out of other pneumonia pathologies like SARS, etc. Thus, we also need more support on pathology understanding to emphasize the most correlated features in the preprocessing phase and augmentations. In a clinical perspective, I think it also helps if we tell how CT experts make their decisions.

**Comment by edgarriba** [comment]
just noticed that the images come in a range between ~ +- 1000

**Comment by Borda** [comment]
it is quite common for medical images as they can be also in tiff with some offset :]

**Comment by edgarriba** [comment]
gotcha. And do we want that for training ? 
https://github.com/mlmed/torchxrayvision/blob/master/torchxrayvision/datasets.py#L47-L51

I think the dataset generator can be improved somehow

**Comment by Borda** [comment]
I think that we shall scale then anyway with the mean and SDT to about (-1, 1) interval

**Comment by edgarriba** [comment]
@Borda sure. Apparently images in this dataset are in png, jpg and jpeg. Some my guess no need apply apply an initial conversion. Please, also check my comment in here: https://github.com/PyTorchLightning/lightning-Covid19/pull/18#discussion_r399652347


![image](https://user-images.githubusercontent.com/5157099/77822161-87248180-70f0-11ea-93db-ffe134b7e8f7.png)

![image](https://user-images.githubusercontent.com/5157099/77822176-a4595000-70f0-11ea-8f8a-b83f4a4b079a.png)

![image](https://user-images.githubusercontent.com/5157099/77822184-b63af300-70f0-11ea-91e3-22c29a8a7776.png)

not sure what would be the best. My guess would the best to analyses the whole image and create some kind of attention to not miss any part.
